### ISSUE TITLE
sources_to_targets returns significantly different results, when direction changed.

### ISSUE THREAD

**Issue by karpinskiJ** [valhalla/valhalla#4921]
Hello!
I'm sending sources_to_targets request to obtain travel_time & disctance between 2 points located in Peru.
 
Request input body:
```
{
  "costing": "auto",
  "sources": [
       {
      "lat":  -12.0665,
      "lon": -75.21226
    }

  ],
  "targets": [

 {
      "lat":  -12.06191,
      "lon": -75.1874
    }

  ]
}
```
Results which I obtained are (travel time and distance make  no sense - checked in google maps ):
```
{
  "id": "my_work_route",
  "algorithm": "costmatrix",
  "units": "kilometers",
  "sources": [
    [
      {
        "lon": -75.21226,
        "lat": -12.0665
      }
    ]
  ],
  "targets": [
    [
      {
        "lon": -75.1874,
        "lat": -12.06191
      }
    ]
  ],
  "sources_to_targets": [
    [
      {
        "distance": 62.238,
        "time": 3984,
        "to_index": 0,
        "from_index": 0
      }
    ]
  ]
}
```
but when I change direction results are reliable:
```
{
  "costing": "auto",
  "sources": [
 {
      "lat":  -12.06191,
      "lon": -75.1874
    }

  ],
  "targets": [


       {
      "lat":  -12.0665,
      "lon": -75.21226
    }
  ]
}
```

```
{
  "id": "my_work_route",
  "algorithm": "costmatrix",
  "units": "kilometers",
  "sources": [
    [
      {
        "lon": -75.1874,
        "lat": -12.06191
      }
    ]
  ],
  "targets": [
    [
      {
        "lon": -75.21226,
        "lat": -12.0665
      }
    ]
  ],
  "sources_to_targets": [
    [
      {
        "distance": 3.321,
        "time": 352,
        "to_index": 0,
        "from_index": 0
      }
    ]
  ]
}
```
Do you have any idea what might be a reason? My only thought is that it's something regarding maps or  land form, but area in nearby seems to be quite flat. There is also output from openstreetmap: https://www.openstreetmap.org/directions?engine=fossgis_valhalla_car&route=-12.066%2C-75.212%3B-12.062%2C-75.187#map=16/-12.06438/-75.20041

**Comment by chrstnbwnkl** [comment]
Yeah we were having the same conversation over at https://github.com/gis-ops/docker-valhalla/issues/168. Doing connection checks from both directions could give you a different output, but that would likely just point to another underlying bug IMO :smile:

**Comment by chrstnbwnkl** [comment]
My goodness, I just tried this out locally and can reproduce it (without Docker). That's a really strange route... will have to investigate this

**Comment by karpinskiJ** [comment]
Changing source_to_targets algorithm to "traveltimdistancematrix" helps, but it's not the answer for the root cause. I tried to visualize this route, but unfortunately when using /route endpoint it does work correctly in both direction.

**Comment by chrstnbwnkl** [comment]
@Trietes intuition seems to be correct by the way, the public instance does the reverse connection checks, I'm assuming it's related. 

Just to make clear what we're talking about (blue is from the public instance, red is local with default config):

![image](https://github.com/user-attachments/assets/da30f380-7ed6-4f71-ad77-f55bebc55144)

**Comment by karpinskiJ** [comment]
@chrstnbwnkl  could you please share request you send to obtain above shapes?

**Comment by chrstnbwnkl** [comment]
You can use `"shape_format": "polyline6"` to obtain the shapes for costmatrix results.

**Comment by nilsnolde** [comment]
we should really do https://github.com/valhalla/valhalla/issues/5256, seems that solved this particular issue
